pennylane/fermi/fermionic.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The Fermionic representation classes."""
+from copy import copy
+
+
+class FermiWord(dict):
+    """Immutable dictionary used to represent a Fermi word, a product of fermionic creation and
+    annihilation operators, associating wires with their respective operators. Can be constructed
+    from a standard dictionary.
+
+    >>> w = FermiWord({(0, 0) : '+', (1, 1) : '-'})
+    >>> w
+    '0+ 1-'
+    """
+
+    def __init__(self, operator):
+        self.sorted_dic = dict(sorted(operator.items()))
+        super().__init__(operator)
+
+    @property
+    def wires(self):
+        """Return wires in a FermiWord."""
+        return [i[1] for i in self.sorted_dic.keys()]
```

Comment:
Should we consider calling this property `sorted_wires` or `all_wires` or something similar? On operators wires usually returns a list of the wires the operator acts on, without duplication. We certainly want to have this property, but based on the rest of PennyLane, I would expect `.wires` to return `[0, 1]` for `"0+ 1+ 0- 1-"`, for example.

Reply:
Yes, it makes more sense to return a set of wires not a list.